aspnetcore/blazor/forms-validation.md

```diff
@@ -296,12 +376,69 @@ public class ShipDescription
 }
 ```
 
-::: moniker-end
+### Enable the submit button based on form validation
+
+To enable and disable the submit button based on form validation:
+
+* Use the form's `EditContext` to assign the model when the component is initialized.
+* Validate the form in the context's `OnFieldChanged` callback to enable and disable the submit button.
+
+```razor
+<EditForm EditContext="@editContext">
+    <DataAnnotationsValidator />
+    <ValidationSummary />
+
+    ...
+
+    <button type="submit" disabled="@formInvalid">Submit</button>
+</EditForm>
+
+@code {
+    private Starship starship = new Starship();
+    private bool formInvalid = true;
```

Comment:
```suggestion
    private bool formValid;
```

Reply:
Personal preference, but it's easier to think in terms of positives

Reply:
Because the attribute is `disabled`, I felt it was simpler as `formInvalid` ... i.e., if the form is invalid, it's true, and the form is disabled (`disabled="true"`).